gnovm/stdlibs/std/emit_event.go

```diff
@@ -46,14 +47,20 @@ func attrKeysAndValues(attrs []string) ([]gnoEventAttribute, error) {
 }
 
 type gnoEvent struct {
-	Type       string              `json:"type"`
+	MsgIdx     int                 `json:"msg_idx"`
 	PkgPath    string              `json:"pkg_path"`
 	Func       string              `json:"func"`
+	Type       string              `json:"type"`
 	Attributes []gnoEventAttribute `json:"attrs"`
 }
 
 func (e gnoEvent) AssertABCIEvent() {}
 
+func (e gnoEvent) SetMsgIdx(msgIdx int) interface{} {
+	e.MsgIdx = msgIdx
+	return e
+}
```

Comment:
Why this instead of `e.MsgIdx = msgIdx` when needed?

Reply:
If what you're suggesting is `e.MsgIdx = msgIdx` without ` return e`, vm was panicking if function is like `func (e *gnoEvnet)`.